src/spetlrtools/diagrams/DiagramParser.py

```diff
@@ -0,0 +1,77 @@
+import xml.etree.ElementTree as ET
+from itertools import chain
+from typing import Dict, List, Optional
+
+from spetlrtools.diagrams.Edge import Edge
+from spetlrtools.diagrams.HTMLStripper import HTMLStripper
+
+
+class DiagramNode:
+    def __init__(self, details):
+        self.id = details["id"]
+        for key in ["value", "label"]:
+            try:
+                self.label = HTMLStripper.strip(details[key])
+                break
+            except KeyError:
+                continue
+        else:
+            self.label = ""
+
```

Comment:
If the loop hits the `except KeyError: continue` - will it jump down to the else statement afterwards?

(I try to figure out if there is a possibility that self.label is not setted.

Reply:
I think it works, but I changed it to make it clearer.